src/Buffer.hpp

```diff
@@ -126,3 +131,22 @@ T BufferPopOne(Buffer<T> *buffer)
     buffer->m_Size = size - 1;
     return buffer->m_Storage[size - 1];
 }
+
+template <typename T>
+bool BufferContains(Buffer<T>* buffer, const T& element)
+{
+    for (int i=0; i!=buffer->m_Size;i++)
+        if (buffer->m_Storage[i] == element)
+            return true;
+    return false;
+}
+
+
+template <typename T, typename U>
+void BufferAppendOneIfNotPresent(Buffer<T> *buffer, MemAllocHeap *heap, U elem)
+{
+    if (BufferContains(buffer, (T)elem))
+        return;
+    T *dest = BufferAlloc(buffer, heap, 1);
+    *dest = (T)elem;
+}
```

Comment:
I can't see any uses of these, are BufferAppendOneIfNotPresent + BufferContains dead code?

Reply:
No, check DagDerivedCompiler.cpp, and Driver.cpp. Github has the unfortunate functionality to hide large files in the diff by default.

Reply:
Sorry, my mistake. Github doesn't expand large diffs by default...